spec/dummy/db/schema.rb

```diff
@@ -116,6 +116,28 @@
     name: :bigint_custom_id_int_lists_b,
     values: [3, 4]
 
+  create_list_partition :bigint_int_list_date_range_subpartitions, partition_key: :id do |t|
+    t.timestamps null: false, precision: nil
+  end
+
+  create_list_partition_of \
+    :bigint_int_list_date_range_subpartitions,
+    name: :bigint_int_list_date_range_subpartitions_a,
+    values: [1, 2],
+    partition_type: :range,
+    partition_key: :created_at
+
+  create_range_partition_of \
+    :bigint_int_list_date_range_subpartitions_a,
+    name: :bigint_int_list_date_range_subpartitions_a_1,
+    start_range: Time.now - 1.day,
+    end_range: Time.now + 10.days
+
+  create_list_partition_of \
+    :bigint_int_list_date_range_subpartitions,
+    name: :bigint_int_list_date_range_subpartitions_b,
+    values: [3, 4]
+
```

Comment:
Should we add some integration tests for hash partitioning? Or maybe you have and I missed that? I guess part of the problem is that we don't want to run the migration for PG 10.

Reply:
Yes, I added a spec `spec/integration/migration_postgres_11_plus_spec.rb`. When CI runs the suite on Postgres 11 and 12 it will run those tests. Works perfectly with your circleci setup that tests the gem on multiple PG versions... it was super helpful.

There is also a spec  `spec/integration/migration_postgres_10_spec.rb` which only runs in Postgres 10 and tests the errors being raised if you try to perform an unsupported action.